src/lib/repo.rs

```diff
@@ -1,8 +1,10 @@
 use anyhow::Result;
+use std::path::Path;
 use std::path::PathBuf;
 
 pub trait MailRepo {
     fn search_message(&self, query: &str) -> Result<Vec<PathBuf>>;
+    fn search_message_with_prefix(&self, query: &str, prefix: &Path) -> Result<Vec<PathBuf>>;
```

Comment:
I think it's better to keep this trait unmodified and apply the filtering after invoking `search_message`. This approach will result in less code, although it may use slightly more memory, and it avoids the need to modify `DummyRepo`.

Reply:
Do you mean that you want the method stripped, or that I implement `search_message_with_prefix` as part of the trait, using `search_message` ?

Reply:
I'd say keep it out of the trait, since the filtering can be implemented as post-processing step (`filter`) of `search_message`